### PR TITLE
Feat/add regexp sqlite

### LINKED ISSUE #564: Snowflake REGEXP function supported?
Hi,

Is the [Snowflake REGEXP function](https://docs.snowflake.com/en/sql-reference/functions/regexp.html) supported?  I see a `regex` function in `pypika`:

~~~python
customers = Tables('customers')
q = Query.from_(customers).select(
    customers.id,
    customers.fname,
    customers.lname,
).where(
    customers.lname.regex(r'^[abc][a-zA-Z]+&')
)
~~~

Which creates:

~~~sql
SELECT id,fname,lname FROM customers WHERE lname REGEX '^[abc][a-zA-Z]+&';
~~~

Is there a way with a `dialect` to change REGEX to REGEXP?

thx

### PR DESCRIPTION
**Pull request by lyz-code**
Closes #564

The changes in the order of the imports were done by `isort`, if you want I can restore the order.

Also changed the version of black from `stable` to `master` to fix the CI

### CODE REVIEW

File: .github/workflows/black.yml [hunk 1]
```diff
@@ -8,4 +9,4 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
-      - uses: psf/black@stable
+      - uses: psf/black@master
```
Comment (gl3nn) [inline .github/workflows/black.yml:12]: Why is this changed?

Comment (gl3nn) [reply .github/workflows/black.yml:12]: Oh because of "Error: Unable to resolve action `psf/black@stable`, unable to find version `stable`". I think it would be better to pin a version instead of master? Not sure.

Comment (lyz-code) [reply .github/workflows/black.yml:12]: Because black changed their main branch and broke all the CIs of the world.

Check the output of the job of commit a5931d7  -> https://github.com/kayak/pypika/runs/2277687520

Comment (gl3nn) [reply .github/workflows/black.yml:12]: Yeah I noticed. If using master is the recommended approach then I'm fine with it.

**Review by gl3nn — APPROVED** [review]
